apps/src/dance/DanceAiClient.ts

```diff
@@ -0,0 +1,101 @@
+import UntypedCachedPalettes from '@cdo/static/dance/ai/model/cached-spacy-palette-map.json';
+import UntypedCachedBackgrounds from '@cdo/static/dance/ai/model/cached-spacy-background-map.json';
+import UntypedCachedForegrounds from '@cdo/static/dance/ai/model/cached-spacy-foreground-map.json';
+
+// Custom types for vector mapping JSONs to play nice with Typescript
+type EmojiAssociation = {
+  association: number[];
+};
+
+type CachedMapping = {
+  emojiAssociations: {[key: string]: EmojiAssociation};
+  output: string[];
+};
+
+const CachedPalettes: CachedMapping = UntypedCachedPalettes;
+const CachedBackgrounds: CachedMapping = UntypedCachedBackgrounds;
+const CachedForegrounds: CachedMapping = UntypedCachedForegrounds;
+
+// Output: JSON containing {backgroundColor:string, backgroundEffect:string, foregroundEffect:string}
+// ToDo: Break down into helper functions for each step rather than repeat code for each field
```

Comment:
NIT: Please make a task for each TODO

Reply:
Should this be in JIRA/Trello?

Reply:
Will be performing a larger reformatting to simplify substeps as described in this trello task: https://trello.com/c/Xt0uC2ai